tests/unit/modules/test_x509.py

```diff
@@ -186,6 +187,360 @@ def test_create_certificate(self):
+pNSx2dU1uJ89YKFlYLo1QcEk4rJ2fdIjarV0kuNCY3rC8jYUp9BpAU5Z6p9HKeT1
+2rTPH81JyjbQDR5PyfCyzYOQtpwpB4zIUUK/Go7tTm409xGKbbUFugJNgQIDAQAB
+AoGAF24we34U1ZrMLifSRv5nu3OIFNZHyx2DLDpOFOGaII5edwgIXwxZeIzS5Ppr
+yO568/8jcdLVDqZ4EkgCwRTgoXRq3a1GLHGFmBdDNvWjSTTMLoozuM0t2zjRmIsH
+hUd7tnai9Lf1Bp5HlBEhBU2gZWk+SXqLvxXe74/+BDAj7gECQQDRw1OPsrgTvs3R
+3MNwX6W8+iBYMTGjn6f/6rvEzUs/k6rwJluV7n8ISNUIAxoPy5g5vEYK6Ln/Ttc7
+u0K1KNlRAkEAx34qcxjuswavL3biNGE+8LpDJnJx1jaNWoH+ObuzYCCVMusdT2gy
+kKuq9ytTDgXd2qwZpIDNmscvReFy10glMQJAXebMz3U4Bk7SIHJtYy7OKQzn0dMj
+35WnRV81c2Jbnzhhu2PQeAvt/i1sgEuzLQL9QEtSJ6wLJ4mJvImV0TdaIQJAAYyk
+TcKK0A8kOy0kMp3yvDHmJZ1L7wr7bBGIZPBlQ0Ddh8i1sJExm1gJ+uN2QKyg/XrK
+tDFf52zWnCdVGgDwcQJALW/WcbSEK+JVV6KDJYpwCzWpKIKpBI0F6fdCr1G7Xcwj
+c9bcgp7D7xD+TxWWNj4CSXEccJgGr91StV+gFg4ARQ==
+-----END RSA PRIVATE KEY-----
+'''
+        # Issue the CA certificate (self-signed)
+        ca_cert = x509.create_certificate(text=True,
+                                          signing_private_key=ca_key,
+                                          CN='Redacted Root CA',
+                                          O='Redacted',
+                                          C='BE',
+                                          ST='Antwerp',
+                                          L='Local Town',
+                                          Email='certadm@example.org',
+                                          basicConstraints="critical CA:true",
+                                          keyUsage="critical cRLSign, keyCertSign",
+                                          subjectKeyIdentifier='hash',
+                                          authorityKeyIdentifier='keyid,issuer:always',
+                                          days_valid=3650,
+                                          days_remaining=0)
+
+        not_before_str = 'this is an intentionally wrong format'
+
+        # Try to sign a new server certificate with the wrong date
+        try:
+            x509.create_certificate(text=True,
+                                    signing_private_key=ca_key,
+                                    signing_cert=ca_cert,
+                                    CN='Redacted Normal Certificate',
+                                    O='Redacted',
+                                    C='BE',
+                                    ST='Antwerp',
+                                    L='Local Town',
+                                    Email='certadm@example.org',
+                                    basicConstraints="critical CA:false",
+                                    keyUsage="critical keyEncipherment",
+                                    subjectKeyIdentifier='hash',
+                                    authorityKeyIdentifier='keyid,issuer:always',
+                                    not_before=not_before_str,
+                                    days_remaining=0)
+            self.fail('Certificate generation must fail as not_before is wrong')
+        except Exception as e:
```

Comment:
You should use `with self.assertRaises(...)` with a narrow exception type - not `except Exception`. *Especially* when testing, you only want to catch the very specific error that you care about. You can find more information in the Python docs.

Reply:
Hey @waynew, thanks for the tip. It looks way better the way you suggested. I also ended up using `self.assertRaisesRegex` instead so I could check for the exception message as well.